### DISCUSSION TITLE
Pessimistic locking with r2dbc

### DISCUSSION THREAD

**Discussion by vu3mmg** [pgjdbc/r2dbc-postgresql#476] [General]
Hi,

Could you please help me to get an example for Pessimistic locking with r2dbc .
Or could we use plain SQL statements such as 
```
begin;
select * from foo where quatloos = 100 for update;
update foo set feens = feens + 1 where quatloos = 100;
commit;
```

This is to take care a scenario to update account balance for a transaction system .

regards
mahesh

**Comment by mp911de** [comment]
R2DBC is just the technical interfacing towards Java. Beyond that, this driver is able to send SQL through the Postgres wire protocol so it runs the SQL you provide to it.

**Reply by vu3mmg** [reply]
Thank you . So we can use all types of native locks provided by postgress .

**Reply by mp911de** [reply]
Yes, exactly.

**Reply by AmadeAli1** [reply]
Como usar bloqueio em postgres??
